modules/ExperimentBrowser/src/client/ExperimentBrowser.coffee

```diff
@@ -156,10 +156,208 @@ class ExperimentSimpleSearchController extends AbstractFormController
 	events:
 		'keyup .bv_experimentSearchTerm': 'updateExperimentSearchTerm'
 		'click .bv_doSearch': 'handleDoSearchClicked'
+		'click .bv_downloadFiles': 'downloadFiles'
 
 	render: =>
 		$(@el).empty()
 		$(@el).html @template()
+	
+	downloadFiles: =>
+		#extract the current search term in the experiment browser
+		experimentSearchTerm = $.trim(@$(".bv_experimentSearchTerm").val())
+		
+		#get the text value in the filter input text field
+		filterSearchTerm = $(".dataTables_filter input").val()
+
+		#replace "-" in filterSearchTerm with spaces (for filtering later on)
+		filterSearchTerm = filterSearchTerm.replaceAll("-", " ")
+
+		#create an array to store all experiment files
+		allExperimentFiles = []
+
+		#search for experiments
+		$.ajax
```

Comment:
I think there should be a way that we can avoid re-doing the search. Sometimes these searches can be quite slow, so it is worth investing a little effort here to seeing if we can avoid re-doing that.

I think we can get the list of Experiments from the ExperimentSummaryTable's `collection` attribute. Accessing that might require you moving this function onto the ExperimentBrowserController.
From "within" the ExperimentBrowserController, `@experimentSummaryTable.collection` should give the list of experiments.

I feel strongly that there should be a way to reuse the filtering capability built into the DataTables widget. You've done something clever here by reimplementing the filtering logic, but I'm concerned about subtle bugs where the filtering you've implemented doesn't quite match the filtering built into DataTables. It'd be much safer if we can access the DataTables object on the frontend and get the list of filtered rows out of it.

Reply:
I think that makes a lot of sense, I see how the current system is fragile and could be broken especially if the DataTable is modified or updated in the future. I wasn't aware of how the experiment data could be accessed that way, will try to re-do this using that instead!